android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/State.kt

```diff
@@ -116,6 +122,18 @@ internal fun State.next(event: Event): State? = when (this) {
             is Event.Account.BeginPairingFlow -> State.Active(ProgressState.BeginningAuthentication)
             else -> null
         }
+        is AccountState.Authenticating -> when (event) {
+            is Event.Progress.AuthData -> {
+                if (event.authData.state == this.accountState.state) {
```

Comment:
I am a bit skeptical about the idea of introducing verification logic into the state transition matrix as we could do the same check before emitting the `AuthData` event, to verify if the event belongs to the existing account. I guess most of my skepticism is coming from the fact that there is no other example of any validation in the `state.next()` function and because `AuthData` event is sent just from one place, that has access to the state to boot. Wonder what you think about it.

Reply:
The main motivation is that I want the app-services code to perform that check and the simplest way to achieve that was putting it inside the state.  However, I realize now that the android-components state doesn't need to change -- I'll undo that.